Sources/Common/DataModel/DataSet/index.js

```diff
@@ -57,6 +28,87 @@ function vtkDataSet(publicAPI, model) {
     }
   });
 
+  //------------------------------------------------------------------------------
+  // Compute the data bounding box from data points.
+  publicAPI.computeBounds = () => {
+    if (
+      (model.modifiedTime &&
+        model.computeTime &&
+        model.modifiedTime > model.computeTime) ||
+      !model.computeTime
+    ) {
+      const points = publicAPI.getPoints();
+      if (points?.getNumberOfPoints()) {
+        // Compute bounds from points
+        const bounds = vtkBoundingBox.INIT_BOUNDS.slice();
+        const numberOfPoints = points.getNumberOfPoints();
+        for (let i = 0; i < numberOfPoints; i++) {
+          const pt = points.getPoint(i);
+          vtkBoundingBox.addPoint(bounds, pt[0], pt[1], pt[2]);
+        }
+        model.bounds = bounds;
+      } else {
+        model.bounds = vtkMath.createUninitializedBounds();
+      }
+      // Update computeTime
+      model.computeTime = Date.now();
+    }
+  };
+
+  /**
+   * Returns the bounds as [xmin, xmax, ymin, ymax, zmin, zmax]
+   */
+  publicAPI.getBounds = () => {
```

Comment:
There should be `getBounds()` and `getBoundsByReference()`. Both methods should be created automatically by `macro.getArray(["bounds"], 6);`, and overriden with `publicAPI.getBounds = macro.chain(() => publicAPI.computeBounds, publicAPI.getBounds);` and `publicAPI.getBoudnsByReference = macro.chain(() => publicAPI.computeBounds(), publicAPI.getBoundsByReference);`